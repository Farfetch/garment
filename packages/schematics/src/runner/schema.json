{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsSchematicSchema",
  "title": "Schematic Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The package name",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the package?"
    }
  },
  "required": []
}
