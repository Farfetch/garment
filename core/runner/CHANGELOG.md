# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.14.0](https://github.com/Farfetch/garment/compare/v0.13.14...v0.14.0) (2020-12-15)


### Features

* Improve cache subscriptions ([#27](https://github.com/Farfetch/garment/issues/27)) ([77bb25a](https://github.com/Farfetch/garment/commit/77bb25acf000319efb05d6e76417aad256042fb6))





## [0.13.13](https://github.com/Farfetch/garment/compare/v0.13.12...v0.13.13) (2020-09-29)

**Note:** Version bump only for package @garment/runner





## [0.13.11](https://github.com/Farfetch/garment/compare/v0.13.10...v0.13.11) (2020-09-14)


### Bug Fixes

* Wrong baseDir in a watch mode ([#19](https://github.com/Farfetch/garment/issues/19)) ([95893ec](https://github.com/Farfetch/garment/commit/95893ec7f5f2ad9b469ca746e66dee595ce49ec1))





## 0.13.8 (2020-07-15)

**Note:** Version bump only for package @garment/runner





## 0.13.7 (2020-07-08)

**Note:** Version bump only for package @garment/runner





## 0.13.6 (2020-07-08)

**Note:** Version bump only for package @garment/runner
