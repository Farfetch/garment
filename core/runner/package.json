{
  "name": "@garment/runner",
  "license": "MIT",
  "version": "0.14.1",
  "main": "lib/index.js",
  "files": [
    "lib",
    "schemas"
  ],
  "dependencies": {
    "@garment/logger": "^0.14.0",
    "@garment/schema-validator": "^0.14.0",
    "@garment/watcher": "^0.14.0",
    "@garment/workspace": "^0.14.1",
    "@types/fs-extra": "^5.0.5",
    "fs-extra": "8.1.0",
    "mustache": "^3.0.1"
  }
}
