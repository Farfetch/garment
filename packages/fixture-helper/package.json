{
  "name": "@garment/fixture-helper",
  "license": "MIT",
  "version": "0.16.0",
  "main": "lib/index.js",
  "files": [
    "lib"
  ],
  "dependencies": {
    "@garment/logger": "^0.16.0",
    "@garment/runner": "^0.16.0",
    "@garment/workspace": "^0.16.0",
    "fs-extra": "8.1.0",
    "globby": "10.0.1",
    "strip-ansi": "^5.2.0",
    "tempy": "0.3.0"
  },
  "devDependencies": {
    "is-valid-path": "^0.1.1",
    "normalize-path": "^3.0.0"
  }
}
