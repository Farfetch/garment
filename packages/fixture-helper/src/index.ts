export * from './executeRunner';
export * from './initFixtureHelper';
export * from './resolver';
import normalizePath = require('normalize-path');
import isValidPath = require('is-valid-path');

const containsValidPath = (value: string): boolean =>
  value.split(/\s/).some((s) => isValidPath(s));

export const normalizeContainedPath = (value: string): string => {
  return value
    .split(/(\s)/)
    .map((s) => {
      if (isValidPath(s)) {
        return normalizePath(s);
      } else {
        return s;
      }
    })
    .join('');
};

export function replaceTestPath<T>(
  obj: T,
  testPath: string,
  replaceWith = '/test_path'
): T {
  const normalizedTestPathRegEx = new RegExp(normalizePath(testPath), 'g');
  return JSON.parse(
    JSON.stringify(obj, (_, value) =>
      typeof value === 'string' && containsValidPath(value)
        ? normalizeContainedPath(value).replace(
            normalizedTestPathRegEx,
            replaceWith
          )
        : value
    )
  );
}
