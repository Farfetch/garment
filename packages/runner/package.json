{
  "name": "@garment/runner",
  "license": "MIT",
  "version": "0.18.0",
  "main": "lib/index.js",
  "files": [
    "lib",
    "schemas"
  ],
  "dependencies": {
    "@garment/logger": "^0.18.0",
    "@garment/schema-validator": "^0.18.0",
    "@garment/watcher": "^0.18.0",
    "@garment/workspace": "^0.18.0",
    "@types/fs-extra": "^5.0.5",
    "fs-extra": "8.1.0",
    "mustache": "^3.0.1"
  }
}
