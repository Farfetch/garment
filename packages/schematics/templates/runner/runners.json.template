{
  "runners": {
    "default": {
      "handler": "./lib/default",
      "schema": "./lib/default/schema.json",
      "description": ""
    }
  }
}
