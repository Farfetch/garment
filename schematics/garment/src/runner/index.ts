import { strings } from '@angular-devkit/core';
import {
  apply,
  chain,
  mergeWith,
  move,
  Rule,
  SchematicContext,
  SchematicsException,
  Tree,
  url
} from '@angular-devkit/schematics';
import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
import {
  addPackageJsonDependencies,
  addProjectToGarmentJson
} from '@garment/schematics-utils';
import * as Path from 'path';
import { applyTemplates } from '../utils/applyTemplates';

export interface PackageSchematicOptions {
  name: string;
  packageName?: string;
  directory: string;
  skipGarmentJson?: boolean;
}

export default function(options: PackageSchematicOptions): Rule {
  return (_tree: Tree, context: SchematicContext) => {
    const { name } = options;

    if (!options.name) {
      throw new SchematicsException('name option is required.');
    }

    const directory = 'plugins';

    const dashedName = strings.dasherize(name);
    const projectName = 'runner-' + dashedName;

    const packagePath = Path.join(directory, projectName);

    context.addTask(
      new NodePackageInstallTask({
        workingDirectory: packagePath,
        packageManager: 'yarn'
      })
    );

    let source = apply(url('../../templates/runner'), [
      applyTemplates(
        {
          name,
          dot: '.',
          ...strings
        },
        { interpolationStart: '___', interpolationEnd: '___' }
      ),
      move(packagePath)
    ]);

    return chain([
      mergeWith(source),
      addPackageJsonDependencies(
        `${packagePath}/package.json`,
        'core/runner/package.json',
        'utils/fixture-helper/package.json'
      ),
      addProjectToGarmentJson({
        garmentJsonPath: '/garment.json',
        name: projectName,
        path: packagePath,
        extendProjects: ['tspackage', 'copy-other-files']
      })
    ]);
  };
}
