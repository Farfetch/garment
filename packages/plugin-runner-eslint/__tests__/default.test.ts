import handler from '../src';
import { initFixtureHelper, executeRunner2 } from '@garment/fixture-helper';
import { file } from '@garment/runner';

const { initFixture, clean } = initFixtureHelper(module, {
  tempDir: __dirname + 'tmp__'
});

afterAll(clean);

const files = [
  file.text('index.js', 'const a = 10'),
  file.text('foo.js', 'var a = "Im still using vars";')
];

test('Emits errors and warnings', async () => {
  const cwd = await initFixture('basic');

  const result = await executeRunner2(
    handler,
    {},
    {
      cwd,
      files
    }
  );

  expect(result).toMatchSnapshot();
});

test('Emits fixed files', async () => {
  const cwd = await initFixture('basic');

  const result = await executeRunner2(
    handler,
    { fix: true },
    {
      cwd,
      files,
      outputDir: '/some/output/dir'
    }
  );

  expect(result).toMatchSnapshot();
});

test('Takes into account .eslintignore', async () => {
  const cwd = await initFixture('ignore');

  const result = await executeRunner2(
    handler,
    {},
    {
      cwd,
      files
    }
  );

  expect(result).toMatchSnapshot();
});
