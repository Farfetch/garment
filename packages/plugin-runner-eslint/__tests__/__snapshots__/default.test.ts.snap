// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emits errors and warnings 1`] = `
Object {
  "collectedOutput": Array [],
  "dependsOnFile": Array [],
  "isLongRunning": false,
  "logs": Array [
    Object {
      "args": Array [
        "Linting files with ESLint",
      ],
      "level": "info",
    },
    Object {
      "args": Array [
        "
/test_path/test/index.js
  1:13  error  Insert \`;⏎\`        prettier/prettier
  1:13  error  Missing semicolon  semi
  
",
      ],
      "level": "error",
    },
    Object {
      "args": Array [
        "
/test_path/test/foo.js
  1:9  error  Replace \`\\"Im·still·using·vars\\";\` with \`'Im·still·using·vars';⏎\`  prettier/prettier
  1 error and 0 warnings potentially fixable with the \`--fix\` option.
",
      ],
      "level": "error",
    },
    Object {
      "args": Array [
        "
/test_path/test/foo.js
  1:1  warning  Unexpected var, use let or const instead  no-var
  1 error and 0 warnings potentially fixable with the \`--fix\` option.
",
      ],
      "level": "warn",
    },
  ],
  "onDestroyHandler": undefined,
  "shouldWatchDependencies": false,
}
`;

exports[`Emits fixed files 1`] = `
Object {
  "collectedOutput": Array [
    Object {
      "data": "const a = 10;
",
      "path": "index.js",
      "type": "text",
    },
    Object {
      "data": "let a = 'Im still using vars';
",
      "path": "foo.js",
      "type": "text",
    },
  ],
  "dependsOnFile": Array [],
  "isLongRunning": false,
  "logs": Array [
    Object {
      "args": Array [
        "Linting files with ESLint",
      ],
      "level": "info",
    },
  ],
  "onDestroyHandler": undefined,
  "shouldWatchDependencies": false,
}
`;

exports[`Takes into account .eslintignore 1`] = `
Object {
  "collectedOutput": Array [],
  "dependsOnFile": Array [],
  "isLongRunning": false,
  "logs": Array [
    Object {
      "args": Array [
        "Linting files with ESLint",
      ],
      "level": "info",
    },
    Object {
      "args": Array [
        "
/test_path/test/index.js
  1:13  error  Insert \`;⏎\`        prettier/prettier
  1:13  error  Missing semicolon  semi
  
",
      ],
      "level": "error",
    },
  ],
  "onDestroyHandler": undefined,
  "shouldWatchDependencies": false,
}
`;
